src/uint/rand.rs

```diff
@@ -108,42 +108,50 @@ pub(super) fn random_mod_core<T>(
     modulus: &NonZero<T>,
     n_bits: u32,
 ) where
-    T: AsMut<[Limb]> + ConstantTimeLess + Zero,
+    T: AsMut<[Limb]> + AsRef<[Limb]> + ConstantTimeLess + Zero + core::fmt::Debug,
```

Comment:
What is this bound change for?

Reply:
You mean `AsRef<[Limb]>`?

It comes from a few lines below:

```rust
let hi_word_modulus = modulus.as_ref().as_ref()[n_limbs - 1].0;
```

Here, `modulus` is a `&NonZero<T>` so I can't use `as_mut()` to get to the limbs. Do you have a better suggestion?

Removed the `Debug` bound, that is leftover debug code.

Reply:
I actually meant `Debug`